src/connmgr/connection.rs

```diff
@@ -4583,18 +4580,19 @@ where
         let finished = {
             loop {
                 match resp_body.try_recv(body_buf.write_buf())? {
-                    RecvStatus::Complete(finished, written) => {
-                        body_buf.write_commit(written);
+                    RecvStatus::Complete(finished, size) => {
+                        body_buf.write_commit(size);
 
                         break finished;
                     }
-                    RecvStatus::Read((), written) => {
-                        body_buf.write_commit(written);
+                    RecvStatus::Read((), size) => {
+                        body_buf.write_commit(size);
 
-                        if written == 0 {
-                            resp_body.add_to_buffer().await?;
+                        if size == 0 {
+                            return Err(Error::BufferExceeded);
                         }
                     }
+                    RecvStatus::NeedBytes(()) => resp_body.add_to_buffer().await?,
                 }
             }
```

Comment:
This is similar to the code that reads request body bytes from a client into a fixed-size buffer, except this time it's for reading response body bytes from a server.

This means the only change in behavior is what it does when the output buffer is full. Previously, it would try to read more bytes into the input buffer, either hanging the connection, hanging the thread (busylooping), or eventually returning `BufferExceeded` if the input buffer filled up also. Now, it immediately returns `BufferExceeded`. So there should be no performance impact during regular operation, just an earlier error when the output is full and avoiding hangs.